src/Modules/UngradedModules/DirectSum.jl

```diff
@@ -92,7 +81,14 @@ function direct_product(M::ModuleFP{T}...; task::Symbol = :prod) where T
   set_attribute!(s, :show => Hecke.show_direct_product, :direct_product => M)
   projection_dictionary = IdDict{Int,ModuleFPHom}()
   injection_dictionary = IdDict{Int,ModuleFPHom}()
-  set_attribute!(s, :projection_morphisms => projection_dictionary, :injection_morphisms => injection_dictionary)
+  ranges = sizehint!(Vector{UnitRange{Int}}(), length(M))
+  i=0
+  for x in M
+    j = i + ngens(x)
+    push!(ranges, i+1:j)
+    i = j
+  end
```

Comment:
This might be debateable but I witnessed that using direct sums/products can really slow down performance in some cases. My solution was to also get access to the index ranges of the distinct summands in order to write fast code. I think that storing this information in a separate attribute does not hurt, does it?